### PR TITLE
Use //go:build generate and consolidate fuzz tests into fuzz_test.go

### PR DESCRIPTION
**Pull request by gaissmai**
## Summary by CodeRabbit

- Tests
  - Reorganized test suite: removed many individual fuzz harnesses, consolidated a large comprehensive fuzz suite for CIDR data structures while preserving deterministic edge-case tests.

- Chores
  - Adjusted code-generation workflow and helper scripts; normalized generated-file headers and build-tag behavior.

- New Features
  - Added a precomputed, read-only lookup table to speed hot-path operations.

### CODE REVIEW

File: fuzz_test.go [hunk 1]
```diff
@@ -0,0 +1,1524 @@
...
+		lite := &Lite{}
+
+		// Setup: Insert first half of prefixes
+		halfCount := len(prefixItems) / 2
+		for i := range halfCount {
+			item := prefixItems[i]
+			lite.Modify(item.pfx, func(bool) bool {
+				return false // insert
+			})
+		}
```
[the remark this answers is not in the document]
Comment (gaissmai) [reply fuzz_test.go:339]: @coderabbitai
lerne bitte endlich, dass diese repo go >= 1.24.0 benützt, du musst eben die go.mod lesen bei go projekten

File: internal/lpm/generate_lookuptbl.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	fmt.Fprintf(os.Stdout, "%s ✓ goimports completd\n", INFO)
+
+	// gofumpt
+	gofumpt := exec.Command("gofumpt", "-w", *outFname)
+	fmt.Fprintf(os.Stdout, "%s Running gofumpt on %s\n", INFO, *outFname)
+	if _, err := gofumpt.CombinedOutput(); err != nil {
+		fmt.Fprint(os.Stderr, DIE, err)
+		os.Exit(1)
+	}
+	fmt.Fprintf(os.Stdout, "%s ✓ gofumpt completd\n", INFO)
```
[the remark this answers is not in the document]
Comment (gaissmai) [reply internal/lpm/generate_lookuptbl.go]: done

[the remark this answers is not in the document]
Comment (gaissmai) [reply internal/lpm/generate_lookuptbl.go]: @coderabbitai done

**Comment by gaissmai** [comment]
> `62-77`: **Log formatter output when they fail.**

@coderabbitai thanks, done

File: internal/lpm/generate_lookuptbl.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+		fmt.Fprintf(os.Stderr, "%s %s", DIE, string(out))
+		os.Exit(1)
+	}
+	fmt.Fprintf(os.Stdout, "%s ✓ gofumpt completed\n", INFO)
+
+	fmt.Println()
+}
+
+func genLookupTbl() string {
+	// gen the llokup table
```
[the remark this answers is not in the document]
Comment (gaissmai) [reply internal/lpm/generate_lookuptbl.go]: > _⚠️ Potential issue_ | _🟡 Minor_
> 
> **Fix typo in comment.**
> 
> "llokup" should be "lookup".
> 
> Apply this diff:
> 
> ```diff
> -	// gen the llokup table
> +	// gen the lookup table
> ```
> 
> 📝 Committable suggestion
> 
> 🤖 Prompt for AI Agents

@coderabbitai
kannst du mir die Generierung mit Kommentaren versehen?

**Comment by gaissmai** [comment]
> `9-30`: **Consider using text/template instead of html/template.**

@coderabbitai
ah, sicher, das war ein problem der IDE

**Comment by gaissmai** [comment]
> `46-50`: **Close file on template execution failure.**

@coderabbitao good point
